samples/python/text_generation/README.md

```diff
@@ -226,6 +226,39 @@ Recommended models: meta-llama/Llama-3.2-1B-Instruct, meta-llama/Llama-3.2-8B-In
 **Note:**  
 Structured output enforcement guarantees correct JSON formatting, but does not ensure the factual correctness or sensibility of the content. The model may generate implausible or nonsensical data, such as `{"name": "John", "age": 200000}` or `{"model": "AbrakaKadabra9999######4242"}`. These are valid JSONs but may not make sense. For best results, use the latest or fine-tuned models for this task to improve the quality and relevance of the generated output.
 
+
+### 12. Tool Calling with Structural Tags Sample (`structural_tags_generation`)
+- **Description:**
+  Structural tags is a technique that allows to switch from regular sampling to structural output generation and back during the text generation.
+  If during the sampling process the model produces a trigger string, it switches to structured mode and generates output according to the schema defined by the tag. After that the model switches back to regular sampling mode.
+  This is useful for generating function calls or other structured outputs that need to follow a specific format.
+
+  This sample demonstrates how to use OpenVINO GenAI to generate structured tool calls from natural language prompts using structural tags. 
+  The model is guided to output function calls in a specific format, enabling integration with external tools: 
+  - Weather API 
+  - Currency exchange APIs
```

Comment:
can you share http links to these external tools?

Reply:
The sample shows only a part of the pipeline and how to generate tool calls, and does not contain actual tool calls. Tool calling is part of the ReAct sample.